### PR TITLE
[ISSUE #8622] 移除 environment 概念

### PR DESCRIPTION
**Pull request by onewe**
移除 environment 概念使用 NProperties 代替, 详情见:  https://www.yuque.com/docs/share/e9fcd9a3-f3fb-496d-9b8b-bf4b88db4df0?# 《Nacos-client 模块参数改造》

Related #8622.

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [x] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

**Review by KomachiSion — CHANGES_REQUESTED** [review]

File: client/src/main/java/com/alibaba/nacos/client/env/NProperties.java [hunk 1]
```diff
@@ -16,12 +16,17 @@
...
+
 /**
- * nacos env interface.
+ * NProperties interface.
  *
  * @author onewe
  */
-public interface NacosEnvironment {
+@SuppressWarnings("PMD.ClassNamingShouldBeCamelRule")
+public interface NProperties {
```
Comment (KomachiSion) [inline client/src/main/java/com/alibaba/nacos/client/env/NProperties.java:27]: What's the meanings of `N`? Nacos or other?

Comment (onewe) [reply client/src/main/java/com/alibaba/nacos/client/env/NProperties.java:27]: The `N` means `Nacos`

Comment (onewe) [reply client/src/main/java/com/alibaba/nacos/client/env/NProperties.java:27]: Or change from the `NProperties` to another names? I have no idea what name is better.

Comment (KomachiSion) [reply client/src/main/java/com/alibaba/nacos/client/env/NProperties.java:27]: Just Nacos client properties？

Comment (onewe) [reply client/src/main/java/com/alibaba/nacos/client/env/NProperties.java:27]: Yes? it's only client properties currently. What else does it have any properties?

**Review by KomachiSion — APPROVED** [review]

**Pull request merged**